builder/quoter.go

```diff
@@ -0,0 +1,42 @@
+package builder
+
+import (
+	"strings"
+)
+
+// Quoter returns safe and valid SQL strings to use when building a SQL text.
+type Quoter interface {
+	// ID quotes identifiers such as schema, table, or column names.
+	// ID does not operate on multipart identifiers such as "public.Table",
+	// it only operates on single identifiers such as "public" and "Table".
+	ID(name string) string
+
+	// Value quotes database values such as string or []byte types as strings
+	// that are suitable and safe to embed in SQL text. The returned value
+	// of a string will include all surrounding quotes.
+	//
+	// If a value type is not supported it must panic.
+	Value(v interface{}) string
+}
+
+// SqlQuoter is default implementation of Quoter interface.
+type SqlQuoter struct {
```

Comment:
maybe the name of this struct can be `Quote`, and the name file is `quote.go` because other structs (except buffer) in this package have that convention